pyogrio/_io.pyx

```diff
@@ -1873,13 +1819,89 @@ def ogr_write(
             CSLDestroy(layer_options)
             layer_options = NULL
 
+    ogr_dataset_out[0] = ogr_dataset
```

Comment:
Is there a reason not to set the pointer instead? (untested)

 ```
*ogr_dataset_out = ogr_dataset
```

Reply:
Yeah, because that's one of the few differences in syntax between cython and C/C++, cython doesn't support the unary `*` (see third bullet point at https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#differences-between-c-and-cython-expressions)

Reply:
Right!  Thanks for the explanation.